pkg/upstream/cluster/host.go

```diff
@@ -35,28 +35,30 @@ import (
 
 // simpleHost is an implement of types.Host and types.HostInfo
 type simpleHost struct {
-	hostname      string
-	addressString string
-	clusterInfo   atomic.Value // store types.ClusterInfo
-	stats         types.HostStats
-	metaData      api.Metadata
-	tlsDisable    bool
-	weight        uint32
-	healthFlags   *uint64
+	hostname                string
+	addressString           string
+	clusterInfo             atomic.Value // store types.ClusterInfo
+	stats                   types.HostStats
+	metaData                api.Metadata
+	tlsDisable              bool
+	weight                  uint32
+	healthFlags             *uint64
+	lastHealthCheckPassTime time.Time
 }
 
 func NewSimpleHost(config v2.Host, clusterInfo types.ClusterInfo) types.Host {
 	// clusterInfo should not be nil
 	// pre resolve address
 	GetOrCreateAddr(config.Address)
 	h := &simpleHost{
-		hostname:      config.Hostname,
-		addressString: config.Address,
-		stats:         newHostStats(clusterInfo.Name(), config.Address),
-		metaData:      config.MetaData,
-		tlsDisable:    config.TLSDisable,
-		weight:        config.Weight,
-		healthFlags:   GetHealthFlagPointer(config.Address),
+		hostname:                config.Hostname,
+		addressString:           config.Address,
+		stats:                   newHostStats(clusterInfo.Name(), config.Address),
+		metaData:                config.MetaData,
+		tlsDisable:              config.TLSDisable,
+		weight:                  config.Weight,
+		healthFlags:             GetHealthFlagPointer(config.Address),
+		lastHealthCheckPassTime: time.Now(),
```

Comment:
It's unnecessary to initialize the `lastHealthCheckPassTime`? Only need to assign a value when the health check changes.

Reply:
Compatible with XDS case. If the host is just discovered, then we need to discover its time as the basis for the slow start weight change. If the last health check pass time of discovered host is zero, we need an other field to record the time when discovered.

Reply:
This will cause all hosts to trigger the slow start strategy at the first time, so it could only take effect if the health flag changes before avoid the issue. Do you think Its ok?

Reply:
I agree with your opinion. 

I just thought about it, maybe the health check pass time of the newly added host can be set in `transferHostSetStates`. 

This will only take effect when a new host is dynamically discovered through xds.

Reply:
Fixed :)